webapp/src/modules/mapBuilder.js

```diff
@@ -269,6 +269,59 @@ export default class MapBuilder {
       });
   }
 
+  setLocateControl() {
+    // Adds locate button to center map on location when clicked
+    const LocateControl = this._l.Control.extend({
+      options: {
+        position: "bottomright",
+        link: undefined,
+        center: () => {
+          this.centerLocation({ ...this._lastLocation, targetZoom: 15 });
+        },
+      },
+      onAdd() {
+        const container = leaflet.DomUtil.create(
+          "div",
+          "leaflet-control-locate leaflet-bar leaflet-control"
+        );
+        const link = leaflet.DomUtil.create(
+          "a",
+          "leaflet-bar-part leaflet-bar-part-single",
+          container
+        );
+        this.options.link = link;
+        link.href = "#";
+        link.title = "Locate me";
+        link.setAttribute("role", "button");
+        leaflet.DomUtil.create("div", "bi bi-geo-fill", link);
+        leaflet.DomEvent.on(
+          this.options.link,
+          "click",
+          () => this.options.center(),
+          this
+        );
+        leaflet.DomEvent.on(this.options.link, "dblclick", (ev) => {
+          leaflet.DomEvent.stopPropagation(ev);
+        });
+        return container;
+      },
+      onRemove() {
+        leaflet.DomEvent.off(
+          this.options.link,
+          "click",
+          () => this.options.center(),
+          this
+        );
+        leaflet.DomEvent.off(this.options.link, "dblclick", (ev) => {
+          leaflet.DomEvent.stopPropagation(ev);
+        });
+      },
+    });
+
+    this._l.control.locate = () => new LocateControl();
```

Comment:
Avoid these sorts of mutations (this and next line) within methods, they are really hard to follow and compose. See newest commit for an alternative.